homeassistant/components/onvif/camera.py

```diff
@@ -108,56 +136,112 @@ def __init__(self, hass, config):
+
+        self._devicemgmt = self._camera.create_devicemgmt_service()
+
+        _LOGGER.debug("Retrieving current camera date/time")
+
+        system_date = dt.datetime.utcnow()
+        cdate = self._devicemgmt.GetSystemDateAndTime().UTCDateTime
+        cam_date = dt.datetime(cdate.Date.Year, cdate.Date.Month,
+                               cdate.Date.Day, cdate.Time.Hour,
+                               cdate.Time.Minute, cdate.Time.Second)
+
+        _LOGGER.debug("Camera date/time: %s",
+                      cam_date)
+
+        _LOGGER.debug("System date/time: %s",
+                      system_date)
+
+        dt_diff = cam_date - system_date
+        dt_diff_seconds = dt_diff.total_seconds()
+
+        if dt_diff_seconds > 5:
+            _LOGGER.warning("The date/time on the camera is '%s', "
+                            "which is different from the system '%s', "
+                            "this could lead to authentication issues",
+                            cam_date,
+                            system_date)
+
+        _LOGGER.debug("Obtaining input uri")
+
+        await self.async_obtain_input_uri()
+
+        _LOGGER.debug("Setting up the ONVIF PTZ service")
+
+        self._ptz_service = self._camera.create_ptz_service()
+
+        _LOGGER.debug("Completed set up of the ONVIF camera component")
+
+    async def async_obtain_input_uri(self):
         """Set the input uri for the camera."""
         from onvif import exceptions
+
         _LOGGER.debug("Connecting with ONVIF Camera: %s on port %s",
                       self._host, self._port)
 
         try:
-            profiles = self._media_service.GetProfiles()
+            _LOGGER.debug("Retrieving profiles")
+
+            media_service = self._camera.create_media_service()
+
+            profiles = media_service.GetProfiles()
+
+            _LOGGER.debug("Retrieved '%d' profiles",
+                          len(profiles))
 
             if self._profile_index >= len(profiles):
                 _LOGGER.warning("ONVIF Camera '%s' doesn't provide profile %d."
                                 " Using the last profile.",
                                 self._name, self._profile_index)
                 self._profile_index = -1
 
-            req = self._media_service.create_type('GetStreamUri')
+            _LOGGER.debug("Using profile index '%d'",
+                          self._profile_index)
 
-            # pylint: disable=protected-access
-            req.ProfileToken = profiles[self._profile_index]._token
-            uri_no_auth = self._media_service.GetStreamUri(req).Uri
+            _LOGGER.debug("Retrieving stream uri")
+
+            req = media_service.create_type('GetStreamUri')
+            req.ProfileToken = profiles[self._profile_index].token
+            req.StreamSetup = {'Stream': 'RTP-Unicast',
+                               'Transport': {'Protocol': 'RTSP'}}
+
+            uri_no_auth = media_service.GetStreamUri(req).Uri
```

Comment:
Should this also be awaited since it's I/O?  Not 100% sure how `zeep[async]` handles these requests.